### PR TITLE
Changes to remove MCT and add ESMF mapping along with other cleanup

### PR DESCRIPTION
**Pull request by mvertens**
This PR does the following:
- removes **all MCT** references from the code and replaces them with ESMF routehandles and mapping calls
- major changes to RtmMod.F90 along with other code cleanup described below
- **RtmVar**
  - Now contains new ESMF data types needed for the MOSART mapping
```F90
       type(ESMF_Field)       , public :: srcField
       type(ESMF_Field)       , public :: dstField
       type(ESMF_RouteHandle) , public :: rh_dnstream
       type(ESMF_RouteHandle) , public :: rh_direct
       type(ESMF_RouteHandle) , public :: rh_eroutUp

```
- **RtmMod:** 
  -  now have  two new init phases for mosart. The first init phase is now called MOSART_init1 and replaces Rtmini. This has mostly what was there before but moves the creation of all routehandles to the second init phase - MOSART_init2 which must be called after the mesh has been read in. Also - moved the section of code for MOSART_init2 to be right below the section for MOSART_init1.
  - removed the mapping for Smatp_dnstrm  since it was not used and there is no reason to create a map that is not needed. The associated code that was commented out for this has also been removed.
  - renamed RtmRun to MOSART_run
  - new indentation
-  **MOSART_physics.F90**
   - now using the computed routehandle rh_eroutUp
   - new indentation
- masterproc -> mainproc

This resolves issue #6 

Testing:
- verified that the following test gave bfb results when compared to noresm2_1_alpha01a
SMS_D_Ld1.f19_f19_mtn14.2000_DATM%GSWP3v1_CLM50%SP_SICE_SOCN_MOSART_SGLC_SWAV.betzy_intel
- verified that the following test passed successfully ERP_Ld3.f19_f19_mtn14.2000_DATM%GSWP3v1_CLM50%SP_SICE_SOCN_MOSART_SGLC_SWAV.betzy_intel

### CODE REVIEW

File: src/riverroute/RtmMod.F90 [hunk 1]
```diff
@@ -1,2849 +1,2347 @@
...
+
+   ! local (gdc)
+   real(r8), pointer :: evel(:,:)       ! effective tracer velocity (m/s)
+   real(r8), pointer :: flow(:,:)       ! mosart flow (m3/s)
+   real(r8), pointer :: erout_prev(:,:) ! erout previous timestep (m3/s)
+   real(r8), pointer :: eroutup_avg(:,:)! eroutup average over coupling period (m3/s)
+   real(r8), pointer :: erlat_avg(:,:)  ! erlateral average over coupling period (m3/s)
+
+   ! global MOSART grid
+   real(r8),pointer :: rlatc(:)    ! latitude of 1d grid cell (deg)
```
Comment (mvdebolskiy) [inline src/riverroute/RtmMod.F90]: can you add `center` in the comment?

File: src/riverroute/RtmMod.F90 [hunk 2]
```diff
@@ -1,2849 +1,2347 @@
...
+      do_rtmflood = .false.
+      ice_runoff  = .true.
+      finidat_rtm = ' '
+      nrevsn_rtm  = ' '
+      coupling_period   = -1
+      delt_mosart = 3600
+      decomp_option = 'basin'
+      bypass_routing_option = 'direct_in_place'
+      qgwl_runoff_option = 'threshold'
+
```
Comment (mvdebolskiy) [inline src/riverroute/RtmMod.F90:135]: 1. Why default decom_option here is basin shouldn't it be round-robin?
2.  Would we benefit from making `namelist_defaults_mosart.xml` and updating `buildnml`?

File: src/riverroute/RtmMod.F90 (hunk 1 above)
Comment (mvertens) [reply src/riverroute/RtmMod.F90]: Done.

**Comment by mvertens** [comment]
@mvdebolskiy - are you okay with this PR?

**Review by mvdebolskiy — APPROVED** [review]
Looks OK for now, considering we are going to refactor other things anyway.

**Pull request merged**